### PR TITLE
Nuke agent is separete role

### PR DESCRIPTION
**Pull request by lzk228**
## About the PR

Nuke Agent is now a separate role in the character customization menu.

**Media**

- [x] I have added screenshots/videos to this PR showcasing its changes ingame, **or** this PR does not require an ingame showcase
- [x] PR is completely finished and I don't need help to finish it.

**Changelog**

:cl: lzk
- tweak: Nuke agent role is now selected separately in the character customization menu

### CODE REVIEW

**Comment by deltanedas** [comment]
changelog should be more explicit that its adding an opt in to agent, not creating a new antag

**Comment by Scribbles0** [comment]
this pr doesn't touch any of the nukeops spawning code so i'm not sure this works as intended right now

**Comment by deltanedas** [comment]
yeah you need to add a pref list for medic then if i == 1 pick medic from that or commander pref or nukie pref or the tide

**Comment by lzk228** [comment]
> yeah you need to add a pref list for medic then if i == 1 pick medic from that or commander pref or nukie pref or the tide

done

File: Content.Server/GameTicking/Rules/NukeopsRuleSystem.cs [hunk 1]
```diff
@@ -494,52 +499,97 @@ private void OnPlayersSpawning(RulePlayerSpawningEvent ev)
...
                         }
-                        nukeOp = _random.PickAndTake(everyone);
-                        Logger.InfoS("preset", "Insufficient preferred nukeop commanders or nukies, picking at random.");
                     }
                     else
                     {
-                        nukeOp = _random.PickAndTake(prefList);
-                        everyone.Remove(nukeOp);
-                        Logger.InfoS("preset", "Insufficient preferred nukeop commanders, picking at random from regular op list.");
+                    nukeOp = _random.PickAndTake(medPrefList);
```
Comment (deltanedas) [inline Content.Server/GameTicking/Rules/NukeopsRuleSystem.cs:524]: indent for this block messed up

File: Content.Server/GameTicking/Rules/NukeopsRuleSystem.cs [hunk 2]
```diff
@@ -494,52 +499,97 @@ private void OnPlayersSpawning(RulePlayerSpawningEvent ev)
...
+                            everyone.Remove(nukeOp);
+                            Logger.InfoS("preset", "Insufficient preferred nukeop agents, picking at random from regular op list.");
+                        }
+                    }
+                    else
                     {
-                        Logger.InfoS("preset", "Insufficient ready players to fill up with nukeops, stopping the selection");
-                        break;
+                    nukeOp = _random.PickAndTake(medPrefList);
+                    everyone.Remove(nukeOp);
```
Comment (deltanedas) [inline Content.Server/GameTicking/Rules/NukeopsRuleSystem.cs:566]: indent for this block messed up

File: Content.Server/GameTicking/Rules/NukeopsRuleSystem.cs [hunk 3]
```diff
@@ -494,52 +499,97 @@ private void OnPlayersSpawning(RulePlayerSpawningEvent ev)
...
                     everyone.Remove(nukeOp);
                     prefList.Remove(nukeOp);
+                    medPrefList.Remove(nukeOp);
                     Logger.InfoS("preset", "Selected a preferred nukeop commander.");
                 }
             }
             else
             {
-                if (prefList.Count == 0)
+                if (i == 1)
```
Comment (deltanedas) [inline Content.Server/GameTicking/Rules/NukeopsRuleSystem.cs:542]: combine this with the else `else if (i == 1)`

**Review by deltanedas — CHANGES_REQUESTED** [review]

File: Content.Server/GameTicking/Rules/NukeopsRuleSystem.cs (hunk 3 above)
Comment (lzk228) [reply Content.Server/GameTicking/Rules/NukeopsRuleSystem.cs:542]: what? how?

Comment (deltanedas) [reply Content.Server/GameTicking/Rules/NukeopsRuleSystem.cs:542]:
```
if (x)
{
}
else
{
    if (y)
    {
    }
    else
    {
    }
}
```

is the same as

```
if (x)
{
}
else if (y)
{
}
else
{
}
```

Comment (lzk228) [reply Content.Server/GameTicking/Rules/NukeopsRuleSystem.cs]: like in commit c7c385a ?

File: Content.Server/GameTicking/Rules/NukeopsRuleSystem.cs [hunk 4]
```diff
@@ -493,6 +498,48 @@ private void OnPlayersSpawning(RulePlayerSpawningEvent ev)
...
+                            everyone.Remove(nukeOp);
+                            Logger.InfoS("preset", "Insufficient preferred nukeop commander or agents, picking at random from regular op list.");
+                        }
+                    }
+                    else
+                    {
+                        nukeOp = _random.PickAndTake(medPrefList);
+                        everyone.Remove(nukeOp);
+                        prefList.Remove(nukeOp);
+                        Logger.InfoS("preset", "Insufficient preferred nukeop commanders, picking an agent")
```
Comment (deltanedas) [inline Content.Server/GameTicking/Rules/NukeopsRuleSystem.cs:527]: add ; here

File: Content.Server/GameTicking/Rules/NukeopsRuleSystem.cs [hunk 5]
```diff
@@ -502,21 +549,21 @@ private void OnPlayersSpawning(RulePlayerSpawningEvent ev)
...
                     }
                 }
                 else
                 {
-                    nukeOp = _random.PickAndTake(cmdrPrefList);
+                    nukeOp = _random.PickAndTake(medPrefList);
                     everyone.Remove(nukeOp);
                     prefList.Remove(nukeOp);
-                    Logger.InfoS("preset", "Selected a preferred nukeop commander.");
+                    Logger.InfoS("preset", "Selected a preferred nukeop agent.")
```
Comment (deltanedas) [inline Content.Server/GameTicking/Rules/NukeopsRuleSystem.cs:566]: and here, to fix building

**Review by deltanedas — CHANGES_REQUESTED** [review]

**Comment by lzk228** [comment]
spawn for preflist was in `Content.Server/GameTicking/Rules/Configurations/NukeopsRuleConfiguration.cs` ?
is that completed? (apart from not successfuled checks)

**Comment by deltanedas** [comment]
i think the pref logic is good

**Comment by lzk228** [comment]
fucking how

**Pull request merged**